### PR TITLE
Next 14 & app router

### PR DESCRIPTION
**Pull request by dominicwest**
Migrating dashboard pages to nexts app router.

Takes advantage of rendering components on the server (React Server Components: RSC) by default, which allows us to completely scrap getServerSideProps and make requests from the component itself.

Additionally can take advantage of "Server actions" which let you submit forms to a server action that executes server side.

This PR includes:
- Creating a root layout in the admin app dir to wrap all pages in the GDS default layout
- Generic service error page that will catch any server/client side errors
- Generic 404 page
- Migrating admin dashboard page to app dir as a RSC
- Migrating super admin dashboard page to app dir as a RSC

TODO:
- initiate govuk-javascript in app dir

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,7 +47,8 @@
     "semver": "^7.5.2",
     "word-wrap": "^1.2.4",
     "@babel/traverse": "7.23.2",
-    "parse-url": "8.1.0"
+    "parse-url": "8.1.0",
+    "domino": "Haringat/domino#2721294c75334ece635dfc72cad0a3dede7391fb"
```
Comment (dominicwest) [inline package.json:51]: Defo want a better way of fixing this but gap-web-ui's `turndown` dependency uses `domino` which uses some deprecated JS syntax that we dont allow due to using `strict` mode. This commit replaces the deprecated JS syntax

File: packages/admin/src/pages/super-admin-dashboard/Navigation.tsx [hunk 2]
```diff
@@ -1,3 +1,5 @@
+'use client';
```
Comment (dominicwest) [inline packages/admin/src/pages/super-admin-dashboard/Navigation.tsx:1]: Can likely fix this to render on the server if we replace the `userRouter` functionality - just didnt look much into this

File: packages/admin/src/app/dashboard/DashboardBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  };
+
+  return (
+    <>
+      {bannerProps && (
+        <ImportantBanner
+          {...(bannerProps === FAILED ? errorBannerProps : bannerProps)}
+        />
+      )}
+    </>
```
Comment (ConorFayleTCO) [inline packages/admin/src/app/dashboard/DashboardBanner.tsx:48]: would it not be better to do something like
```suggestion
    if (statusArray.includes(FAILED))
      return {
        bannerHeading: 'Something went wrong while transferring your data.',
        bannerContent: (
          <p className="govuk-body">
            Please get in contact with our support team at{' '}
            <a
              className="govuk-notification-banner__link"
              href="mailto:findagrant@cabinetoffice.gov.uk"
            >
              findagrant@cabinetoffice.gov.uk
            </a>
            {'.'}
          </p>
        ),
        isSuccess: false,
      };
    if (statusArray.includes(SUCCEEDED))
      return {
        bannerHeading:
          'Your data has been successfully added to your One Login account.',
        isSuccess: true,
      };

    return null;
  };

  const bannerProps = getBannerProps(searchParams);

  return (
    <>
      {bannerProps && (
        <ImportantBanner
          {...bannerProps}
        />
      )}
    </>
```

Comment (dominicwest) [reply packages/admin/src/app/dashboard/DashboardBanner.tsx:48]: most definitely

File: packages/admin/src/pages/super-admin-dashboard/Navigation.tsx (hunk 2 above)
Comment (dominicwest) [reply packages/admin/src/pages/super-admin-dashboard/Navigation.tsx]: Resolved - cannot use a component that needs app router or pages router stuff in both app router & pages router, so opted for duplicating the component

**Pull request closed without merging**